hspec-core/src/Test/Hspec/Core/Example.hs

```diff
@@ -40,12 +40,64 @@ import           Test.Hspec.Core.Util
 import           Test.Hspec.Core.QuickCheck.Util (liftHook)
 import           Test.Hspec.Core.Example.Location
 
--- | A type class for examples
+-- | A type class for examples, that is to say, test bodies as used in
+-- `Test.Hspec.Core.Spec.it` and similar functions.
 class Example e where
+  -- | Argument type passed to function-like instances of `Example`.
```

Comment:
I think this phrasing is a little confusing - it's not about functions so much as "the argument/extra data required to run an `e` as an `Example`"

So if `type Arg X = ()`, that's just a way of saying "I have no requirements"

Similarly we have in codebase `instance Example (SqlPersistT IO ()) where type Arg (SqlPersistT IO ()) = SqlBackend`. Which *is* function-like, in some sense: `SqlPersistT m a ~ ReaderT SqlBackend m a ~ SqlBackend -> m a` - but I think "function-like" makes a reader think more of things like `instance (Example r) => Example (a -> r)` pattern, allowing

```haskell
before mkThing do
  it "has stuff" $ \thing -> do
    thing `shouldBe` "whatever"
```

Possibly:

```suggestion
  -- | Argument type required to run the `Example` test.
```